### PR TITLE
Disallow annotations with mutable objects

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
$title. This was previously allowed because the std lib used such annotations.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/24246

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by MaryamZi** [comment]
Although https://github.com/ballerina-platform/ballerina-standard-library/issues/74 was fixed, seems like there are other mutable fields in HTTP annotations now - https://github.com/ballerina-platform/ballerina-standard-library/issues/4300.

File: tests/jballerina-unit-test/src/test/resources/test-src/annotations/annots_with_invalid_type.bal [hunk 1]
```diff
@@ -20,3 +20,13 @@ type VAL true;
 type trueArray VAL[];
 
 annotation trueArray v2 on type;
+
+annotation record {| Foo f; |} v3 on type;
```
Comment (lochana-chathura) [inline tests/jballerina-unit-test/src/test/resources/test-src/annotations/annots_with_invalid_type.bal:24]: I personally prefer the current way (We use this style in nBallerina too). However, our formatter would format like this though. 
```suggestion
annotation record {|Foo f;|} v3 on type;
```

**Review by lochana-chathura — APPROVED** [review]
Looks Good.

File: tests/jballerina-unit-test/src/test/resources/test-src/annotations/annots_with_invalid_type.bal (hunk 1 above)
Comment (MaryamZi) [reply tests/jballerina-unit-test/src/test/resources/test-src/annotations/annots_with_invalid_type.bal:24]: Shall we leave as is (the string representation also uses spaces)? 

I think it helps readbility for inclusive/exclusiveness too

```
annotation record {lName name;} a on type;
// vs
annotation record {|lName name;|} b on type;
```

vs

```
annotation record { lName name; } a on type;
// vs
annotation record {| lName name; |} b on type;
```

I'm not sure why we don't have spaces here, may have taken a similar approach to mapping constructors, but they aren't comparable IMO.

**Comment by MaryamZi** [comment]
Changing PR state to draft since stdlib needs to be fixed.

**Pull request merged**